general/Random.h

```diff
@@ -0,0 +1,147 @@
+ */
+template <typename RealType = double>
+RealType GetNormal(RealType mean, RealType stddev)
+{
+    if(!_init) {
+        throw std::logic_error("RandomNumber not yet seeded!");
+    }
+
+    std::normal_distribution dist(mean, stddev);
+    return dist(_randomEngine);
+}
+
+/**
+ * Returns an integer random number of a uniform distribution with min (\p min) and max (\p max).
+ * @pre \p min <= \p max
+ * @tparam IntType
+ * @param min min of the uniform distribution
+ * @param max max of the uniform distribution
+ * @return random number of a uniform distribution with given parameters
+ */
+template <typename IntType = int>
+IntType GetUniformInt(IntType min, IntType max)
+{
+    if(!_init) {
+        throw std::logic_error("RandomNumber not yet seeded!");
+    }
+
+    std::uniform_int_distribution dist(min, max);
+    return dist(_randomEngine);
+}
+
+/**
+ * Returns an real random number of a uniform distribution with min (\p min) and max (\p max).
+ * @pre \p min <= \p max
+ * @tparam RealType
+ * @param min min of the uniform distribution
+ * @param max max of the uniform distribution
+ * @return random number of a uniform distribution with given parameters
+ */
+template <typename RealType = double>
+RealType GetUniformReal(RealType min, RealType max)
+{
+    if(!_init) {
+        throw std::logic_error("RandomNumber not yet seeded!");
+    }
+
+    std::uniform_real_distribution dist(min, max);
+    return dist(_randomEngine);
+}
+
+/**
+ * Returns an real random number of a triangular distribution with min (\p min), max (\p max) and peak (\p peak).
+ * Between \param min and \param peak the probability is increasing linearly and from \param peak to \p max it is decreasing linearly.
+ * @pre \p min <= \p peak <= \p max
+ * @tparam RealType
+ * @param min min of the triangular distribution
+ * @param max max of the triangular distribution
+ * @param peak peak of the triangular distribution
+ * @return random number of a triangular distribution with given parameters
+ */
+template <typename RealType = double>
+RealType GetTriangular(RealType min, RealType max, RealType peak)
+{
+    if(!_init) {
+        throw std::logic_error("RandomNumber not yet seeded!");
+    }
+
+    std::vector<RealType> i{min, peak, max};
+    std::vector<RealType> w{0, 1, 0};
+    std::piecewise_linear_distribution<RealType> dist{i.begin(), i.end(), w.begin()};
+    return dist(_randomEngine);
+}
+
+/**
+ * Returns a random number of a lognormal distribution with mean (\p mean) and standard deviation (\p stddev).
+ * @tparam RealType
+ * @param mean mean of the normal distribution
+ * @param stddev standard deviation of the normal distribution
+ * @return random number of a normal distribution with given parameters
+ */
+template <typename RealType = double>
+RealType GetLogNormal(RealType mean, RealType stddev)
+{
+    if(!_init) {
+        throw std::logic_error("RandomNumber not yet seeded!");
```

Comment:
Maybe we should use `assert` here. If you use an uninitialized random number generator it's more like a programming error ...